### PR TITLE
[Hub Generated] Publish private branch 'users/prjayasw/agrifoodrp'

### PR DESCRIPTION
**Pull request by manikprabhu1208**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/published%2Fmanikprabhu1208%2Fazure-rest-api-specs-pr%2Fusers%2Fprjayasw%2Fagrifoodrp...main/).
## ARM API Information (Control Plane)

Please review the last 3 commits of this branch

1. 745222b24a5e2ccf465b0c5f312c01142eee9087
2. 62391ff3e13856622d29590115ecff87ee191c46
3. 2447e0cadbde62d18684869118b51acf11c50c3c

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  4. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  5. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  6. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by manikprabhu1208** [comment]
The PR is created based on the updates in the private branch. The updates in the PR has already been reviewed and approved with this PR [Azure/azure-rest-api-specs-pr/8500](https://github.com/Azure/azure-rest-api-specs-pr/pull/8500)

File: custom-words.txt
Comment (TimLovellSmith) [inline custom-words.txt:14]: Is this some special jargon? I think the usual word is not Abstractive but 'abstract' (n., v., adj.), or 'abstracted' (v, p.t.), 'abstractly' (adv.) etc.

File: custom-words.txt
Comment (TimLovellSmith) [inline custom-words.txt:885]: You're adding lots of words that are already in the list. Can you just revert this file and see if spellcheck still passes?

Comment (manikprabhu1208) [reply custom-words.txt:885]: I was getting merge confict, so I pulled main branch into mine, and all these words came up.

Comment (TimLovellSmith) [reply custom-words.txt:885]: Can you just revert this file and see if spellcheck still passes?

**Comment by TimLovellSmith** [comment]
@manikprabhu1208 please also fill out the survey form in the PR description.

**Comment by manikprabhu1208** [comment]
> @manikprabhu1208 please also fill out the survey form in the PR description.
Done

**Comment by visingla-ms** [comment]
>     "version": "2021-09-01-preview"

Why so old dated version?

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:6 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
Hi, you need to get Breaking Changes reviewed by Breaking Reviews panel, before proceeding further with swagger review. Please remove ARMChangesRequested flag once breaking changes approved or resolved.

**Comment by manikprabhu1208** [comment]
> Hi, you need to get Breaking Changes reviewed by Breaking Reviews panel, before proceeding further with swagger review. Please remove ARMChangesRequested flag once breaking changes approved or resolved.

We already got breaking change approval in private repo

**Comment by visingla-ms** [comment]
> > ```
> > "version": "2021-09-01-preview"
> > ```
> 
> Why so old dated version?
> 
> Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:6 in [c46f7d7](https://github.com/Azure/azure-rest-api-specs/commit/c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00). [](commit_id = [c46f7d7](https://github.com/Azure/azure-rest-api-specs/commit/c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00), deletion_comment = False)

can you please share?

**Comment by manikprabhu1208** [comment]
> > ```
> > "version": "2021-09-01-preview"
> > ```
> 
> Why so old dated version?
> 
> Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:6 in [c46f7d7](https://github.com/Azure/azure-rest-api-specs/commit/c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00). [](commit_id = [c46f7d7](https://github.com/Azure/azure-rest-api-specs/commit/c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00), deletion_comment = False)

We have all our changes in this api-version

**Comment by visingla-ms** [comment]
>             "description": "Success",

It should be "Updated" and as it will go to SDK, you can add more details in description. (applicable to all places).

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:49 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
>             "description": "Success"

204 means, resource was already deleted or not found.

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:150 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
>             "name": "$maxPageSize",

You should use standard OData query parameters if possible: $top in this case.

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:211 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
>             "name": "$skipToken",

Do you mean OData $skip here?

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:221 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
>             "name": "extensionIds",

Try to use OData $filter for filtering purposes e.g. $filter = "extensionId eq a or extensionId eq b" etc.

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:191 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
>               "$ref": "#/definitions/ArmAsyncOperation"

Name should be as per actual operation e.g. it could be FarmBeatOperationResult or something.

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:757 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
>     "/providers/Microsoft.AgFoodPlatform/farmBeatsExtensionDefinitions/{farmBeatsExtensionId}": {

Are these not scoped to subscriptions? tenant-level resources?

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:339 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           },

Use it from common-types: [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v4/privatelinks.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v4/privatelinks.json).

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:896 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           },

Use from common types.

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:955 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "nextLinkName": null

Please evaluate if this list might need pagination, if not now, but in future.

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1094 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "nextLinkName": null

Pagination is not needed?

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1142 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

File: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json
Comment (visingla-ms) [inline specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1220]: >v3 [](http://example.com/codeflow?start=69&length=2)

Use v4 at al places to be consistent with rest of the swagger.

File: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json
Comment (visingla-ms) [inline specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1244]: >SolutionInstallationRequest [](http://example.com/codeflow?start=37&length=27)

Why not same request and response schema?
SolutionInstallationRequest vs Solution?

File: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json
Comment (visingla-ms) [inline specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1351]: It is good to define it once in parameters, and then refer it everywhere. Helps in future edits and avoiding typos.

File: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json
Comment (visingla-ms) [inline specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1361]: >Success [](http://example.com/codeflow?start=28&length=7)

204 is already deleted or not found.

File: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json
Comment (visingla-ms) [inline specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1519]: >farmBeatsSolutionDefinitions [](http://example.com/codeflow?start=41&length=28)

Are these these tenant-level resources? Why not subscription scoped?

**Comment by visingla-ms** [comment]
>           "x-ms-identifiers": []

The "x-ms-identifiers" should not be empty. Please put unique identifier for items in the array. See: [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-identifiers](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-identifiers).

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1878 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           ],

ProvisioningState model is not correct. It should have at-least these 3 values: Succeeded, Failed, Canceled. See: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#provisioningstate-property](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#provisioningstate-property).

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1932 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           ],

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#provisioningstate-property](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#provisioningstate-property).

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1975 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

File: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json
Comment (visingla-ms) [inline specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:2161]: As per RPC, the etag should be returned via response header, not via properties. See: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#etags-for-resources](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#etags-for-resources).

File: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json
Comment (visingla-ms) [inline specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:2163]: >"x-ms-azure-resource": true [](http://example.com/codeflow?start=6&length=27)

Not needed if a resource is reference of TrackedResource or ProxyResource.

File: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json
Comment (visingla-ms) [inline specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:2257]: Why eTag as property, not as response header?

File: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json
Comment (manikprabhu1208) [reply specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1519]: These are not subscription scoped. These are like global resources which any customer can install them.

File: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json
Comment (manikprabhu1208) [reply specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1244]: SolutionInstallationRequest is a subset of Solution model. We just wanted to show only the required fields in body.

File: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json
Comment (manikprabhu1208) [reply specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:2161]: we are checking the ETag property on our side and passing it in response headers.

**Comment by manikprabhu1208** [comment]
> > ```
> >         "name": "extensionIds",
> > ```
> 
> You can also use OData $filter for filtering purposes e.g. $filter = "extensionId eq a or extensionId eq b" etc. Same for extensionCategories.
> 
> Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:191 in 78f326f. [](commit_id = [78f326f](https://github.com/Azure/azure-rest-api-specs/commit/78f326f8c1b1097d8db6e8b74743cf1436c716d1), deletion_comment = False)

These changes have already been merged into public repo. When I raised PR these were not there, so they have appeared here. This pr will only add Solutions resource. Extensions resource is already present in public repo.

**Comment by visingla-ms** [comment]
>               "$ref": "#/definitions/ExtensionInstallationRequest"

Why Put is not using same model for Input and Output? ExtensionInstallationRequest and Extension should be same. Any reason why they are not?

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:43 in a9c5245. [](commit_id = a9c524577454b0c8585b6d541874e6ef22ed17f1, deletion_comment = False)

**Comment by rkmanda** [comment]
Should this be 2022-09-01-preview?

---
In reply to: [1332636279](https://github.com/Azure/azure-rest-api-specs/pull/21583#issuecomment-1332636279) [](http://example.com/codeflow?ancestors=1332636279,1332636279)

File: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json
Comment (sjanamma) [reply specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1239]: Request and Response models must be the same for PUt

**Comment by sjanamma** [comment]
>           ],

this needs to be addressed

---
In reply to: [1332780656](https://github.com/Azure/azure-rest-api-specs/pull/21583#issuecomment-1332780656) [](http://example.com/codeflow?ancestors=1332780656,1332780656)

---
Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1932 in c46f7d7. [](commit_id = c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00, deletion_comment = False)

**Comment by sjanamma** [comment]
@manikprabhu1208 Please address all the review comments. This change also requires NewAPIVersion.

**Comment by manikprabhu1208** [comment]
> > ```
> >       ],
> > ```
> 
> this needs to be addressed
> 
> In reply to: [1332780656](https://github.com/Azure/azure-rest-api-specs/pull/21583#issuecomment-1332780656)
> 
> Refers to: specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2021-09-01-preview/agfood.json:1932 in [c46f7d7](https://github.com/Azure/azure-rest-api-specs/commit/c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00). [](commit_id = [c46f7d7](https://github.com/Azure/azure-rest-api-specs/commit/c46f7d758fdbe2cbc27c86377f8bbf72ac76ff00), deletion_comment = False)

This change is already merged in public repo. They are not my changes. My changes are only the last 3 commits and example file changes

**Comment by manikprabhu1208** [comment]
> @manikprabhu1208 Please address all the review comments. This change also requires NewAPIVersion.

We need to add these changes in same api version. We took breaking change approval for it already.

**Review by honghr — APPROVED** [review]

**Pull request merged**
